src/gc/gc.cpp

```diff
@@ -838,7 +831,7 @@ class t_join
             return TRUE;
         }
 
-        if (Interlocked::Decrement(&join_struct.r_join_lock) != (join_struct.n_threads - 1))
+        if (Interlocked::CompareExchange(&join_struct.r_join_lock, 0, join_struct.n_threads) == 0)
```

Comment:
For ARM64 CompareExchange can be significantly faster than Exchange/Decrement.

In this r_join case.  Only the first thread needs to write.  All others can simply read.  First thread writes 0.